### PR TITLE
Re former feature

### PR DESCRIPTION
**Pull request by mahmoud717**
in this project we followed the odin project form instructions

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  end
+
+  def create
+    @user = User.new(user_params)
+    # @user = User.new(username: params[:username], email: params[:email], password: params[:password])
+
+    if @user.save
+      redirect_to new_user_path
+    else
+      redirect_to users_path
```
Comment (maelfosso) [inline app/controllers/users_controller.rb]: That route doesn't exists

Redirect to `new_user_path`

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    @user = User.find(params[:id])
+    if @user.update(user_params)
+      redirect_to users_path
+    else
+      render :edit
+    end
+  end
+
+  def user_params
+    params.require(:user).permit(:username, :email, :password)
```
Comment (maelfosso) [inline app/controllers/users_controller.rb:30]: That route doesn't exists
I got this error when update an user
Screenshot at 2020-09-30 17:43:26

Redirect to `edit_user_path`

File: app/views/users/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<!--<form method="POST" action="/users" accept-charset="UTF-8">
```
Comment (maelfosso) [inline app/views/users/new.html.erb]: it's better to use rails helper `users_path`

File: app/views/users/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    <label for="username">Username</label>
+    <input id="username" name="user[username]" type="text">
+    <label for="email">email</label>
+    <input id="email" name="user[email]" type="email">
+    <label for="password">password</label>
+    <input id="password" name="user[password]" type="password">
+    <input type="submit" value="submit">
+</form>-->
+
+<!--<%= form_tag('/users') do %>
```
Comment (maelfosso) [inline app/views/users/new.html.erb]: it's better to use rails helper `users_path`

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    # @user = User.new(username: params[:username], email: params[:email], password: params[:password])
```
Comment (maelfosso) [inline app/controllers/users_controller.rb]: Remove comments

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,

Thank you for the job done so far
You did an almost good job. Just a few issues to fix.

Kindly read the comments left below to know more about the issues.
Solve them and submit a new code review request

Thanks
Mael FOSSO

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Hello @mahmoud717, @Haywhizzz :wave:. Well done on the job so far:+1: Let's take a look at this again**

:point_right: In your create action, when a user fails to save, it should render :new, not redirect.

_**Optional Improvements**_
1. In the update action, when a user is successfully updated, it should redirect to edit_user_path.
2. You can create a partial for the user form and use the partial in both new and edit views
3. You can display form errors when they exist on your forms.
4. You can validate the email using REGEXP.
5. In future projects add PR descriptions that express work done, not a generic message like you have on this PR.
6. Your README, should be more descriptive. How do I start/use the project? Who are the authors of the project? etc...

_**Kindly ensure that you make the mandatory change and then submit another code review request.**_

**Any questions? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested:exclamation:

**Review by Dandush03 — APPROVED** [review]
# Status: :rocket: Project Approved :green_circle: 
Great work implementing all mandatory changes
### Good luck in your next project
# Happy Coding :tophat: :computer:

**Pull request merged**